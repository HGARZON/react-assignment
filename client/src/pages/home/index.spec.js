import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import HomeView from'./index';

Enzyme.configure({ adapter: new Adapter() })

it('renders without crashing', () => {
    shallow(<HomeView />);
});

describe('<HomeView />', () => {
    const wrapper = shallow(<HomeView />);

    it('should render a class `.header`', () => {
      expect(wrapper.find('.header'));
    });

    /* Snapshot */
    it('renders correctly', () => {
        const component = renderer.create(<HomeView />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
});