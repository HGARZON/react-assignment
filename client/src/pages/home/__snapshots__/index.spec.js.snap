// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeView /> renders correctly 1`] = `
<div
  className="header"
>
  <h1
    className="header__title"
  >
    Rick and Morty Character Information
  </h1>
  <div>
    <img
      alt="Rick and Morty"
      className="header__image"
      src="header.jpg"
    />
  </div>
  <a
    className="header__link-information"
    href="https://rickandmortyapi.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    https://rickandmortyapi.com
  </a>
</div>
`;
